AzureDevOpsPowerShellAPI/Public/Api/Policy/Configuration/Set-AzDoBranchPolicyCommentResolution.ps1

```diff
@@ -64,76 +59,59 @@ function Set-AzDoBranchPolicyCommentResolution {
   )
 
   begin {
-    if (-not($script:header)) {
-
-      try {
-        New-ADOAuthHeader -PAT $PAT -ErrorAction Stop
-      } catch {
-        $PSCmdlet.ThrowTerminatingError($_)
-      }
-    }
+    $result = New-Object -TypeName "System.Collections.ArrayList"
   }
 
-  Process {
-    Write-Debug "CollectionUri: $CollectionUri"
-    Write-Debug "ProjectName: $ProjectName"
-    Write-Debug "RepoName: $RepoName"
-    Write-Debug "branch: $branch"
-    Write-Debug "Required: $Required"
-
-    try {
-      $policyId = Get-BranchPolicyType -CollectionUri $CollectionUri -ProjectName $ProjectName -PAT $PAT -PolicyType "Comment requirements"
-    } catch {
-      throw $_.Exception.Message
-    }
+  process {
 
-    try {
-      $repoId = (Get-AzDoRepo -CollectionUri $CollectionUri -ProjectName $ProjectName -PAT $PAT -RepoName $RepoName).RepoId
-    } catch {
-      throw $_.Exception.Message
+    $params = @{
+      uri     = "$CollectionUri/$ProjectName/_apis/policy/configurations"
+      version = "7.2-preview.1"
+      method  = 'POST'
     }
 
-    $body = @{
-      isEnabled  = $true
-      isBlocking = $Required
-      type       = @{
-        id = $policyId
-      }
-      settings   = @{
-        scope = @(
-          @{
-            repositoryId = $repoId
-            refName      = "refs/heads/$branch"
-            matchKind    = "exact"
-          }
-        )
+    $policyId = (Get-AzDoBranchPolicyType -CollectionUri $CollectionUri -ProjectName $ProjectName -PolicyType "Comment requirements").policyId
+
+    foreach ($name in $RepoName) {
+      $repoId = (Get-AzDoRepo -CollectionUri $CollectionUri -ProjectName $ProjectName -RepoName $name).RepoId
+
+      $body = @{
+        isEnabled  = $true
+        isBlocking = $Required
+        type       = @{
+          id = $policyId
+        }
+        settings   = @{
+          scope = @(
+            @{
+              repositoryId = $repoId
+              refName      = "refs/heads/$branch"
+              matchKind    = "exact"
+            }
+          )
+        }
       }
-    }
 
-    $params = @{
-      uri         = "$CollectionUri/$ProjectName/_apis/policy/configurations?api-version=7.2-preview.1"
-      Method      = 'POST'
-      Headers     = $script:header
-      body        = $Body | ConvertTo-Json -Depth 99
-      ContentType = 'application/json'
+      if ($PSCmdlet.ShouldProcess($ProjectName, "Create Branch policy named: $($PSStyle.Bold)$name$($PSStyle.Reset)")) {

```

Comment:
Note to self: We have to be a bit careful naming a lot of vars $name, since it's also in the foreach statement.
In this case it won't give problems, but can become confusing for junior developers.